bokehjs/src/lib/models/tools/inspectors/hover_tool.ts

```diff
@@ -495,9 +480,12 @@ export class HoverToolView extends InspectToolView {
     return rows
   }
 
-  _render_template(template: HTMLElement, tooltips: [string, string][], ds: ColumnarDataSource, i: number | ImageIndex, vars: TooltipVars): HTMLElement {
+  _render_template(template: HTMLElement, tooltips: [string, string][], ds: ColumnarDataSource, vars: TooltipVars): HTMLElement {
     const el = template.cloneNode(true) as HTMLElement
 
+    // if we have an image_index, that is what replace_placeholders needs
+    const i = is_undefined(vars.image_index) ? vars.index : vars.image_index
```

Comment:
The trade-off here was to simplify the calling in half a dozen  other places, and allowing.